### PR TITLE
Move the remaining app catalog logic to the app catalog state domain

### PR DESCRIPTION
**Pull request by axbarsan**
Towards giantswarm/giantswarm#10461

Pretty much what the title says.

### CODE REVIEW

File: src/components/Auth/Login.tsx [hunk 1]
```diff
@@ -19,8 +19,7 @@ import { parseErrorMessages } from './parseErrorMessages';
 
 // The props coming from the global state (AKA: `mapStateToProps`)
 interface IStateProps {
-  // For now, until we have a type for it
-  user: Record<string, never>;
+  user: IUser;
```
Comment (axbarsan) [inline src/components/Auth/Login.tsx:22]: Unrelated 😅

File: src/components/Cluster/ClusterDetail/KeypairCreateModal/KeyPairCreateModal.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ const StyledModal = styled(BootstrapModal)`
 `;
 
 interface IKeyPairCreateModalProps {
-  user: Record<string, never>;
+  user: IUser;
```
Comment (axbarsan) [inline src/components/Cluster/ClusterDetail/KeypairCreateModal/KeyPairCreateModal.tsx:38]: Unrelated 😅

File: src/stores/callAPIMiddleware.js [hunk 3]
```diff
@@ -19,30 +19,29 @@ export function callAPIMiddleware({ dispatch, getState }) {
...
-        if (throwOnError) {
-          throw error;
-        }
-
-        // TODO: Remove this, this supports a pattern we want to factor out
-        // eventually.
-        return {
-          error: error.message,
-        };
+    try {
```
Comment (axbarsan) [inline src/stores/callAPIMiddleware.js:22]: The problem here was when the `shouldPerform()` check returned `false`. The result would've been `undefined`, which has no `then()` method, and then resulted in an error.

File: src/stores/callAPIMiddleware.js [hunk 4]
```diff
@@ -19,7 +19,10 @@ export function callAPIMiddleware({ dispatch, getState }) {
       type: types.request,
     });
 
-    return doPerform(getState(), dispatch)
+    const request = doPerform(getState(), dispatch);
+    if (!request) return next(action);
```
Comment (axbarsan) [inline src/stores/callAPIMiddleware.js:23]: This is needed because if we use `shouldPerform()`, and it returns `false`, then the result of `doPerform()` is `undefined`, which is not a promise, so we can't use `.then()` on it.

**Review by ubergesundheit — APPROVED** [review]
Works like a charm
